### PR TITLE
Add Readme

### PR DESCRIPTION
**Pull request by DIRECTcut**
#2 "Setup access to microphone and camera API"

### CODE REVIEW

File: dedicated-examples/mock-camera-microphone-access/README.md [hunk 1]
```diff
@@ -1 +1,29 @@
-# mock camera/microphone access
+# Mock Camera/Microphone Access
+
+This example shows how to set up access to microphone and camera API. This example runs in **Chrome**.
```
Comment (miherlosev) [inline dedicated-examples/mock-camera-microphone-access/README.md:3]: `This example shows how to mock access to microphone`

**Review by miherlosev — CHANGES_REQUESTED** [review]

**Review by VasilyStrelyaev — APPROVED** [review]

**Review by miherlosev — APPROVED** [review]

File: dedicated-examples/mock-camera-microphone-access/README.md [hunk 2]
```diff
@@ -1 +1,29 @@
-# mock camera/microphone access
+# Mock Camera/Microphone Access
+
+This example shows how to mock data for microphone and camera. This example runs in **Chrome**.
```
Comment (dirk-pieterse) [inline dedicated-examples/mock-camera-microphone-access/README.md:3]: This example runs in Chrome and shows how to use mock data for the microphone and camera.

File: dedicated-examples/mock-camera-microphone-access/README.md [hunk 3]
```diff
@@ -1 +1,29 @@
...
+
+**Test Code**: [mock-camera-microphone-access.js](mock-camera-microphone-access.js)
+
+The test page requests a permission to record audio and video. Resulting streams are directed to corresponding `<audio>` and `<video>` elements on the page. When the `<button>` element registers a click, a snapshot of the video stream draws on the `<img>` element if video stream is present.
```
Comment (dirk-pieterse) [inline dedicated-examples/mock-camera-microphone-access/README.md:7]: The test page requests **permission** to record audio and video. **The** resulting streams are directed to **the** corresponding `<audio>` and `<video>` elements on **a** page (or _the page_ if you are talking about a specific page. However, this isn't clear from the context) . When the `<button>` element registers a click, a snapshot of the video stream **is drawn** on the `` element (if **a** video stream is present).

File: dedicated-examples/mock-camera-microphone-access/README.md [hunk 4]
```diff
@@ -1 +1,29 @@
...
+
+The first test function uses [Selector.addCustomDOMProperties](https://devexpress.github.io/testcafe/documentation/reference/test-api/selector/addcustomdomproperties.html) method to read `<img>`'s `.src` property and check whether it holds an image.
```
Comment (dirk-pieterse) [inline dedicated-examples/mock-camera-microphone-access/README.md:9]: The first test function uses **the** [Selector.addCustomDOMProperties](https://devexpress.github.io/testcafe/documentation/reference/test-api/selector/addcustomdomproperties.html) method to read **the** ``'s `.src` property and check whether it **contains** an image.

Comment (dirk-pieterse) [reply dedicated-examples/mock-camera-microphone-access/README.md:9]: You need to work on your **articles**.

File: dedicated-examples/mock-camera-microphone-access/README.md [hunk 5]
```diff
@@ -1 +1,29 @@
...
+
+The second test function uses [Selector.addCustomDOMProperties](https://devexpress.github.io/testcafe/documentation/reference/test-api/selector/addcustomdomproperties.html) method to read `<audio>`'s `.srcObject` property. If audio stream is present, it returns `true`. This value is tested with [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html) method.
```
Comment (dirk-pieterse) [inline dedicated-examples/mock-camera-microphone-access/README.md:11]: same here

Comment (dirk-pieterse) [reply dedicated-examples/mock-camera-microphone-access/README.md:11]: If audio stream is present -> If **an** audio stream is present

Comment (dirk-pieterse) [reply dedicated-examples/mock-camera-microphone-access/README.md:11]: "This value is tested with [t.expect.ok](https://devexpress.github.io/testcafe/documentation/reference/test-api/testcontroller/expect/ok.html) method"? What do you mean 'is tested with'? How can something be tested with something else?

File: dedicated-examples/mock-camera-microphone-access/README.md [hunk 6]
```diff
@@ -1 +1,29 @@
...
+
+`getUserMedia` allows access to the Media Stream API. When called, it spawns a dialog window. Use `--use-fake-ui-for-media-stream` Chrome flag to grant the permission without user input.
```
Comment (dirk-pieterse) [inline dedicated-examples/mock-camera-microphone-access/README.md:13]: - "When called" When what is called? 
- "it spawns"? Why do you use this verb?
- "to grant the permission" Why do you use an article for permission?

File: dedicated-examples/mock-camera-microphone-access/README.md [hunk 7]
```diff
@@ -1 +1,29 @@
...
+
+If your environment lacks audio/video recording hardware, use `--use-fake-device-for-media-stream` to feed a test pattern to the API.
```
Comment (dirk-pieterse) [inline dedicated-examples/mock-camera-microphone-access/README.md:15]: If your environment does not have audio/video recording hardware

File: dedicated-examples/mock-camera-microphone-access/README.md [hunk 8]
```diff
@@ -1 +1,29 @@
...
+
+Chrome doesn't allow Media Stream API calls from insecure origins. To get round this, do either of the following:
```
Comment (dirk-pieterse) [inline dedicated-examples/mock-camera-microphone-access/README.md:17]: Chrome does not allow insecure Media Stream API calls

File: dedicated-examples/mock-camera-microphone-access/README.md [hunk 9]
```diff
@@ -1 +1,29 @@
...
+
+* Run TestCafe locally with a [--hostname](https://devexpress.github.io/testcafe/documentation/reference/configuration-file.html#hostname) CLI parameter:
+
+```sh
+testcafe --hostname localhost "chrome --use-fake-ui-for-media-stream --use-fake-device-for-media-stream" index.js
+```
+
+* [Generate a self-signed certificate](https://devexpress.github.io/testcafe/documentation/guides/advanced-guides/test-https-features-and-http2-websites.html#use-a-self-signed-certificate) and enable HTTPS with [--ssl](https://devexpress.github.io/testcafe/documentation/reference/command-line-interface.html#--ssl-options) CLI parameter. Use the `--ignore-certificate-errors` Chrome flag to ignore certificate warnings.
```
Comment (dirk-pieterse) [inline dedicated-examples/mock-camera-microphone-access/README.md:25]: "with [--ssl] CLI parameter" an? the?

File: dedicated-examples/mock-camera-microphone-access/mock-camera-microphone-access.js [hunk 10]
```diff
@@ -16,7 +16,7 @@ test('take a photo', async t => {
 
 test('check an audio element source', async t => {
     const audioFrame = Selector('audio').addCustomDOMProperties({
-        //this property is true element has source, otherwise false
+        //this property is true if element has a source, otherwise false
```
Comment (dirk-pieterse) [inline dedicated-examples/mock-camera-microphone-access/mock-camera-microphone-access.js:19]: "if element" an? the?

Comment (dirk-pieterse) [reply dedicated-examples/mock-camera-microphone-access/mock-camera-microphone-access.js:19]: Again, you need to work on your **articles**.

**Pull request merged**